src/Faithlife.Build/DotNetClassicTool.cs

```diff
@@ -62,11 +62,14 @@ public sealed class DotNetClassicTool
 		var packagesPath = Environment.GetEnvironmentVariable("NUGET_PACKAGES") ??
 			Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".nuget", "packages");
 
-		// Find a matching package. If using a specific version, there will be exactly one match.
-		// If using floating versions, there may be multiple matches, so take the latest semantic version.
-		var packagePath = Directory.GetDirectories(Path.Combine(packagesPath, packageName.ToLowerInvariant()), packageVersion)
-			.OrderByDescending(fullPath => NuGetVersion.Parse(Path.GetFileName(fullPath)))
-			.FirstOrDefault();
+		var packageDirectories = Directory.GetDirectories(Path.Combine(packagesPath, packageName.ToLowerInvariant()))
+			.Select(fullPath => NuGetVersion.Parse(Path.GetFileName(fullPath)));
+
+		if (!packageDirectories.Any())
+			throw new BuildException($"Missing restored NuGet package: {Path.Combine(packagesPath, packageName.ToLowerInvariant(), packageVersion)}");
+
+		var bestMatch = VersionRange.Parse(packageVersion).FindBestMatch(packageDirectories).OriginalVersion;
```

Comment:
What does `FindBestMatch` return if nothing matches? Or will it even return packages that are completely wrong?